### ISSUE TITLE
releng - doc action caching seems busted

### ISSUE THREAD

**Issue by kapilt** [cloud-custodian/cloud-custodian#8345]
trunk merges are taking 24m on doc builds, should be ~4m with cache.

**Comment by kapilt** [comment]
also affects prs

**Comment by kapilt** [comment]
even on a successful cache restoration, sphinx still sees 1229 source files out of date.

```
myst v0.18.1: MdParserConfig(commonmark_only=False, gfm_only=False, enable_extensions=[], disable_syntax=[], all_links_external=False, url_schemes=('http', 'https', 'mailto', 'ftp'), ref_domains=None, highlight_code_blocks=True, number_code_blocks=[], title_to_header=False, heading_anchors=None, heading_slug_func=None, footnote_transition=True, words_per_minute=200, sub_delimiters=('{', '}'), linkify_fuzzy_links=True, dmath_allow_labels=True, dmath_allow_space=True, dmath_allow_digits=True, dmath_double_inline=False, update_mathjax=True, mathjax_classes='tex2jax_process|mathjax_process|math|output_area')
[90](https://github.com/cloud-custodian/cloud-custodian/actions/runs/4344194084/jobs/7587204590#step:5:91)
building [mo]: targets for 0 po files that are out of date
[91](https://github.com/cloud-custodian/cloud-custodian/actions/runs/4344194084/jobs/7587204590#step:5:92)
building [html]: targets for 1229 source files that are out of date
[92](https://github.com/cloud-custodian/cloud-custodian/actions/runs/4344194084/jobs/7587204590#step:5:93)
updating environment: 0 added, 152 changed, 0 removed
[93](https://github.com/cloud-custodian/cloud-custodian/actions/runs/4344194084/jobs/7587204590#step:5:94)
reading sources... [ 16%] actions .. aws/examples/index
[94](https://github.com/cloud-custodian/cloud-custodian/actions/runs/4344194084/jobs/7587204590#step:5:95)
reading sources... [ 33%] aws/examples/lambdaerrorsnotify .. azure/advanced/azurepolicy
[95](https://github.com/cloud-custodian/cloud-custodian/actions/runs/4344194084/jobs/7587204590#step:5:96)
reading sources... [ 50%] azure/advanced/contribute .. azure/examples/sqldatabaseupdateretentionpolicies-databases
[96](https://github.com/cloud-custodian/cloud-custodian/actions/runs/4344194084/jobs/7587204590#step:5:97)
reading sources... [ 66%] azure/examples/sqldatabasewithpremiumsku-databases .. gcp/examples/dns-managed-zone
[97](https://github.com/cloud-custodian/cloud-custodian/actions/runs/4344194084/jobs/7587204590#step:5:98)
reading sources... [ 83%] gcp/examples/dns-policy .. kubernetes/examples/require-replicas
[98](https://github.com/cloud-custodian/cloud-custodian/actions/runs/4344194084/jobs/7587204590#step:5:99)
reading sources... [100%] kubernetes/examples/restrict-service-account-usage .. tools/omnissm
```

**Comment by kapilt** [comment]
for now in #8341 we're going to temporarily disable the awscc provider in docs, it takes the cold build on docs back down 10m and gives us some time to investigate the sphinx caching issue, note this issues persists across 4 -> 5 migration #8341 wrt to seeing extraneous changed files, could be an algorithm change in now they calculate toctrees.

**Comment by kapilt** [comment]
resolved in #8341 improved how we update the doc cache so it should hopefully not have a drift staleness issue between dependency upgrades, now it just updates every commit to trunk.
